### DISCUSSION TITLE
Lit + Firebase first-class integration (litfire)

### DISCUSSION THREAD

**Discussion by e111077** [lit/lit#3295] [Ideas]
Heya everyone! We've been looking into the possibility of building a Reactive Controller-based integration with Firebase but simply don't have the time. The sketchpad of ideas we have are:

- Reactive Controller implementation
- Composable enough to take advantage of Firebase's module tree shaking
- An API that is idiomatic to web components that is just spankin' awesome DX

Starting the conversation to see if there is interest in this project and who would be willing to team up with us and some people on the Firebase team to help shape and help implement the project.

Please comment if this sounds up your alley!

**Comment by ynxtynxt** [comment]
Really looking forward to this. I've been trying to wrap Firestore resources with class in JS (dunno TS ugh..) and too bad the API is still in a terrible shape. Basically I'd love to have some controls/options over Firestore doc's properties, per-propery observability to invoke host `requestUpdate()` and property convertors (kinda like dom convertor but for security). Been trying to implement `@property()` and currently `@children()` for sub collection and `@ref()` for update dependencies (so one linked resource update can also trigger the components update). Dunno I can give much input (I am super noob) but I really want to use this thing. And it would be nice to allow a transaction mechanism somehow.

**Comment by christophe-g** [comment]
Very intersted  in this. 

> Reactive Controller implementation with Firebase 

This has been our approach at [accessiblesurveys.com](https://accessiblesurveys.com) (and [idata.tools](https://idata.tools)). An old version was published [here](https://github.com/preignition/lit-firebase), but the library has evolved internally quite a bit (integration of reactive controller, binding with firestore, migrated to TS). 

This is now part of our own monorepo, but I can share a more recent version if there is interest. Unfortunately, it is not as clean as I whish: we do what we can with our little resources ; )

**Comment by bivens-dev** [comment]
Also very much interested in this, was something I was looking to explore myself a bit further into the future but I think it's an important part of creating a nice ecosystem for building apps with Lit.

**Comment by petecarapetyan** [comment]
Very interested, minimally skilled. Perhaps I could help consume it as it develops?
